Add and edit seeds on a target

jeremy-liss opened the pull request:
- Add and remove seeds on a target
- Edit a seed
- Add and remove harvest auths on a seed

TODO - when GET Harvest Auth and Auth agent endpoints are complete:
- Display harvest auth info on a seed
- Display Auth Agent info on a harvest auth, using dropdown accordion in the harvest auth modal
- Option to auto search Harvest auths by seed URL

leefrank9527 commented on webcurator-ui/src/views/target/target-tabs/TargetTabPanelGeneral.vue:
It's good to set the selected model for Dropdown:
const selectedAuthorisationOption = ref('Auto');
<Dropdown v-model="selectedAuthorisationOption" :options="['Auto', 'Add Later']"/>

leefrank9527 commented:
1. It's good to set the selected model for Dropdown:
const selectedAuthorisationOption = ref('Auto');
<Dropdown v-model="selectedAuthorisationOption" :options="['Auto', 'Add Later']"/>
2. The code blocks about Groups, Seeds, Archive options can be abstracted and wrapped in components.
3. Duplicated seed urls are allowed, and the operation on the row will be applied to all the duplicated rows, that looks strange. 
4. That would be better to show an error message if it's failed to save the data. A Toast window can be tipped at the corner.
5. The header name is missing for the action column. In the Old UI, there are "View Permission" and "Unlink Permission" actions.

jeremy-liss commented:
Thanks for the feedback @leefrank9527 

1. I've added a model to the dropdown
2. I agree, Seeds and Groups are large and have a lot of functionality specific to them, so I have moved them to their own containers. I think Archive Options is quite a small block, and doesn't have any functions, so I have left it where it is for now
3. I've added a check to see if the seed already exists on the target and an error message if it does
4. I've added error and success toasts on save target. Also removed the status code from the rest api error message to make it a bit more readable to users. Maybe we could look at improving error messages/objects in future
5. Have added the actions column name - we can add the permission buttons when the Harvest Auth and Auth Agent endpoints are done

leefrank9527 commented:
@jeremy-liss The updated could work as expected. For how to present the error messages, it's a public scenario, and we could have a discussion and process it in a dedicated story.

@obrienben  Agree to merge this PR. Please do it when you are free.
